### PR TITLE
Fixes #53: Restore build permissions

### LINKED ISSUE #53: Wrong permissions when extracting archive
We keep running into permission issues after a deployment, and, while permissions are set correctly when creating the archive, it seems that they're not retained when extracting the archive.

Permissions are set correctly here: https://github.com/jalogut/magento2-deployer-plus/blob/master/recipe/magento_2_1/files.php#L20

But are not retained here: https://github.com/jalogut/magento2-deployer-plus/blob/master/recipe/magento_2_2/artifact.php#L30

I think a `-p` flag is needed: https://www.gnu.org/software/tar/manual/html_node/Setting-Access-Permissions.html#SEC82

I just thought I'd post here first before creating a PR, because I find it strange I'm only running into this issue now (after having used this for years).

It could be that I'm only running into this issue now because of different umask settings, and that most writable directories are shared between deployments anyway (and so setting perms once manually would fix it for good). On top of that, some directories don't need to be writable, such as vendor, pub/static (other than [bug #13225](https://github.com/magento/magento2/issues/13225)) and app/etc, unless you're in developer mode and would like to change env.php through the admin).

### PR DESCRIPTION
**Pull request by erfanimani**
See issue #53

### CODE REVIEW

**Comment by erfanimani** [comment]
(As a side-note, this exposed an issue with Gitlab where permissions are to liberal: https://gitlab.com/gitlab-org/gitlab-runner/issues/1736)

File: recipe/magento_2_1/files.php [hunk 1]
```diff
@@ -17,7 +17,7 @@
 task('files:static_assets', '{{bin/php}} {{magento_bin}} setup:static-content:deploy {{languages}} {{static_deploy_options}}');
 task(
     'files:permissions',
-    'cd {{magento_dir}} && chmod -R g+w var vendor pub/static pub/media app/etc && chmod u+x bin/magento'
+    'chmod -R g+w var vendor pub/static pub/media app/etc && chmod u+x bin/magento'
```
Comment (erfanimani) [inline recipe/magento_2_1/files.php:20]: On another note, vendor really shouldn't be writable

File: recipe/magento_2_2/artifact.php [hunk 2]
```diff
@@ -26,7 +26,8 @@
 task('artifact:upload', function () {
     upload(get('artifact_path'), '{{release_path}}');
 });
+
 task('artifact:extract', '
-	tar -xzf {{release_path}}/{{artifact_file}} -C {{release_path}};
+	tar -xzpf {{release_path}}/{{artifact_file}} -C {{release_path}};
```
Comment (osrecio) [inline recipe/magento_2_2/artifact.php:31]: I agree in this case

File: recipe/magento_2_1/files.php (hunk 1 above)
Comment (osrecio) [reply recipe/magento_2_1/files.php:20]: Why you delete `cd {{magento_dir}}` ?

I can agree with vendor, but we follow the permissions guideline from Magento:
https://devdocs.magento.com/guides/v2.3/install-gde/prereq/file-system-perms.html

Comment (erfanimani) [reply recipe/magento_2_1/files.php:20]: Ah yes, I forgot there's a Setup Wizard that lets you install vendor packages using the UI. I'll add this back in.

cd magento_dir is unnecessary (check other commands, also check debug logs and it does cd twice)

**Comment by erfanimani** [comment]
Thanks for reviewing @osrecio, as per my comments I don't think any changes are necessary.

**Comment by osrecio** [comment]
Your PR  is merged @erfanimani 

Release 2.7.0

**Pull request merged**